pkg/authentication/handlers.go

```diff
@@ -17,15 +17,38 @@ import (
 	"github.com/canonical/identity-platform-admin-ui/internal/validation"
 )
 
+const (
+	codeParameter  = "code"
+	stateParameter = "state"
+
+	codeParamNotFoundMsg   = "OAuth2 code not found"
+	stateParamNotFoundMsg  = "OAuth2 state not found"
+	noIDTokenMsg           = "unable to retrieve ID token"
+	nonceCookieNotFoundMsg = "nonce cookie not found"
+	idTokenNonceMatchMsg   = "id token nonce does not match nonce cookie"
+	stateCookieNotFoundMsg = "state cookie not found"
+	stateParamMatchMsg     = "state parameter does not match state cookie"
+)
+
+var (
+	codeParamNotFoundErr   = errors.New(codeParamNotFoundMsg)
+	stateParamNotFoundErr  = errors.New(stateParamNotFoundMsg)
+	noIDTokenErr           = errors.New(noIDTokenMsg)
+	nonceCookieNotFoundErr = errors.New(nonceCookieNotFoundMsg)
+	idTokenNonceMatchErr   = errors.New(idTokenNonceMatchMsg)
+	stateCookieNotFoundErr = errors.New(stateCookieNotFoundMsg)
+	stateParamMatchErr     = errors.New(stateParamMatchMsg)
```

Comment:
why `errors.New` and not `fmt.Errorf`?second one is used more across the codebase

i've got a tiny issue with global vars, as they can be changed randomly in the codebase and we would get unpredictable results

can we instantiate new objects and return them instead?